src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,297 @@
+                                                                       InstructorAttributes instructor) {
+        List<FeedbackQuestionTableRow> copyQuestionRows = new ArrayList<FeedbackQuestionTableRow>();
+        if (instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION)) {
+            for (FeedbackQuestionAttributes question : copiableQuestions) {
+                String courseId = question.courseId;
+                String fsName = question.feedbackSessionName;
+                String qnType = question.getQuestionDetails().getQuestionTypeDisplayName();
+                String qnText = question.getQuestionDetails().questionText;
+                String qnId = question.getId();
+                
+                FeedbackQuestionTableRow row = new FeedbackQuestionTableRow(courseId, fsName, qnType, qnText, qnId);
+                copyQuestionRows.add(row);
+            }
+        }
+        return copyQuestionRows;
+    }
+
+
+    private void buildPreviewForm(FeedbackSessionAttributes feedbackSession,
+                                    List<StudentAttributes> studentList,
+                                    List<InstructorAttributes> instructorList) {
+        previewForm = new FeedbackSessionPreviewForm(feedbackSession.courseId, feedbackSession.feedbackSessionName, 
+                                                     getPreviewAsStudentOptions(studentList), 
+                                                     getPreviewAsInstructorOptions(instructorList));
+    }
+
+
+    private void buildExistingQuestionForm(String feedbackSessionName,
+                                           int questionsSize,
+                                           Map<String, Boolean> questionHasResponses,
+                                           String courseId, FeedbackQuestionAttributes question) {
+        FeedbackQuestionEditForm qnForm = new FeedbackQuestionEditForm();
+        qnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_EDIT));
+        qnForm.setCourseId(courseId);
+        
+        qnForm.setFeedbackSessionName(feedbackSessionName);
+        qnForm.setQuestion(question);
+        qnForm.setQuestionNumberSuffix("-" + question.questionNumber);
+        
+        qnForm.setNumOfQuestionsOnPage(questionsSize);
+        qnForm.setQuestionNumberOptions(getQuestionNumberOptions(questionsSize));
+        
+        FeedbackQuestionFeedbackPathSettings feedbackPathSettings = new FeedbackQuestionFeedbackPathSettings();
+        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings();
+        qnForm.setFeedbackPathSettings(feedbackPathSettings);
+        qnForm.setVisibilitySettings(visibilitySettings);
+        feedbackPathSettings.setGiverParticipantOptions(getParticipantOptions(question, true));
+        feedbackPathSettings.setRecipientParticipantOptions(getParticipantOptions(question, false));
```

Comment:
While it works fine setting `feedbackPathSettings` and `visibilitySettings` into `qnForm` before setting values into each of them, it should make sense to go with the sequence:

1. create them
* set values into them
* set them into the form

rather than:

1. create them
* set them into the form
* set values into them